### ISSUE TITLE
Feature Request: HDR (P010/Rec.2100)

### ISSUE THREAD

**Issue by polyh3dron** [fzwoch/obs-teleport#77]
Please consider adding HDR to this plugin, it would make it the only viable way to game on a PC at 4K in HDR and do dual PC streaming in HDR and/or tonemap on the 2nd PC while recording in HDR on the gaming PC. As it is, it's possible to tonemap to SDR on the gaming PC's output, but this also means any OBS recording on the main PC will also be tonemapped to SDR.

**Comment by fzwoch** [comment]
Current used jpeg stack is 8 bits per pixel only. Same goes for the used image types in Go.

Not sure if its trivial to modify current used jpeg stack to 10, 12, or 16 bits per pixel. If so, one would also have to change quite a portion where the image types are being used..

Anyway, I'm not planning on diving further into that. Someone else has to chime in here.

**Comment by fzwoch** [comment]
Internally I switched to turbojpeg v3 API. There are APIs for 12 bit and 16 bit encodes and decodes. However there seems to be no SIMD support for these bit dephs yet and therefore are in the range of 2x-3x slower compared to 8 bit, so I'm not very keen to try it out atm.

**Comment by Redferne** [comment]
Any updates on supporting P010 HDR? Or are there other means to achieve 4K@60 HDR streaming between two OBS instances on LAN?

**Comment by fzwoch** [comment]
No. Jpeg-turbo seems to be still a magnitude slower in 12-bit compared to 8-bit. You can of course still output OBS in Rec.2020 and keep an 8-bit format...

**Comment by YorVeX** [comment]
> Any updates on supporting P010 HDR? Or are there other means to achieve 4K@60 HDR streaming between two OBS instances on LAN?

It might not seem that way since 4K HDR videos and 4K gaming has been a thing for a longer while now, but even in 2025 that's not an easy thing (at least when not using professional level hardware), but really something for enthusiast level productions only. If that sounds like you, I can see two things you could try.

### SRT

I haven't tried this myself, but if it's OK to use lossy encoding already in the beginning of your production pipeline, you can use SRT to stream from OBS to OBS, both OBS as a sender and as a receiver are explained [here](https://obsproject.com/kb/srt-protocol-streaming-guide). On the upside, when on LAN you can use very high bandwidth settings (500 mpbs and up) so at least you should be able to avoid compression artifacts.
It will not be as comfortable and simple as using a solution like Teleport, but on the upside you can leverage hardware encoding so you don't strain your CPU. Basically whatever you can stream or record directly from OBS should also be possible to be streamed and recorded this way.

Be aware that this means you basically do the same hardware encoding you would do when just streaming/recording directly from your gaming/content system. So you only actually gain something here if the streaming/production machine adds more things to your stream or recording, like computation heavy (shader) effects, live overlays, sound/music, VTuber avatar animation - this load would then be taken off the main machine. Otherwise, at this point just stream/record directly on the main machine.
Also, I am not 100% sure whether OBS also really supports receiving the same formats it can send, but it sure is worth a try.

### Beam

[My Beam plugin](https://github.com/YorVeX/xObsBeam) has the same problem as Teleport, there are no video encoding formats as efficient as those for non-HDR content. However, there you have the option to transmit raw data or to use a generic (not video specific) compression which is data agnostic and therefore can also transmit HDR content, without any loss.

The caveat here is that a standard gigabit LAN won't cut it, not even close. I got [a table here](https://github.com/YorVeX/xObsBeam/wiki/Compression-help) where you can see that 4K P010 HDR at 60 FPS of raw data would need a bandwidth of more than 11 gbps, meaning not even a 10G connection would suffice. Either you set up an expensive 40G network, or you use 10G (cheaper, but still expensive) use the Density compression algorithm and hope that it's enough to keep it within 10G range.
